src/Components/WebAssembly/JSInterop/src/Microsoft.JSInterop.WebAssembly.csproj

```diff
@@ -17,4 +17,8 @@
     <Compile Include="$(SharedSourceRoot)JSInterop\*.cs" />
   </ItemGroup>
 
+  <ItemGroup>
+    <InternalsVisibleTo Include="Microsoft.AspNetCore.Components.WebAssembly" />
```

Comment:
I see this is necessary due to `JSObjectReferenceJsonConverter` being `internal`. However we tried pretty hard to avoid the IVT when originally building this.

Longer term, it would be preferable to make `JSObjectReferenceJsonConverter` be `public`, staying within the `Microsoft.JSInterop.Infrastructure` namespace, because that namespace is intended only for platform authors rather than application authors. It makes sense for it to be public because platform authors really do need to use it if they want to provide a custom concrete `IJSObjectReference` type. If we did this, the IVT would not be needed.

Is your proposal in this PR intended mainly as a minimally-invasive patch, which we'd then follow up on in 6.0 to clean up and eliminate the IVT? If so I can get behind that.

Or if you think it's reasonable to add the extra public type in a 5.x patch let's discuss that. I know we're usually disinclined to expand the API surface in patch releases.

Reply:
Oh now I read your comment above more thoroughly:

> This change is intentionally made using IVTs which makes it more pliable for a 5.0. If this was 6.0 only change, there's a better way to fix this.

So yes it sounds like we have the same opinion about this.